app/scripts/controller-init/defi-positions/defi-positions-controller-init.ts

```diff
@@ -1,30 +1,35 @@
+import { DeFiPositionsController } from '@metamask/assets-controllers';
+import { ControllerInitFunction } from '../types';
 import {
-  DeFiPositionsController,
   DeFiPositionsControllerMessenger,
-} from '@metamask/assets-controllers';
-import { ControllerInitFunction } from '../types';
-import { DeFiPositionsControllerInitMessenger } from '../messengers/defi-positions/defi-positions-controller-messenger';
+  DeFiPositionsControllerInitMessenger,
+} from '../messengers/defi-positions';
 
 export const DeFiPositionsControllerInit: ControllerInitFunction<
   DeFiPositionsController,
   DeFiPositionsControllerMessenger,
   DeFiPositionsControllerInitMessenger
 > = ({ initMessenger, controllerMessenger, getController }) => {
   const getPreferencesController = () => getController('PreferencesController');
+  const getOnboardingController = () => getController('OnboardingController');
 
   const controller = new DeFiPositionsController({
     messenger: controllerMessenger,
     isEnabled: () => {
-      const preferencesController = getPreferencesController();
-      const { useExternalServices } = preferencesController.state;
+      const {
+        state: { useExternalServices },
+      } = getPreferencesController();
+      const {
+        state: { completedOnboarding },
+      } = getOnboardingController();
 
       const state = initMessenger.call('RemoteFeatureFlagController:getState');
 
       const featureFlagForDeFi = Boolean(
         state?.remoteFeatureFlags?.assetsDefiPositionsEnabled,
       );
 
-      return useExternalServices && featureFlagForDeFi;
+      return completedOnboarding && useExternalServices && featureFlagForDeFi;
```

Comment:
Now that it does not start polling on keyring unlock (which occurs before onboarding is finished), this is technically speaking not necessary.

Still, better to have it just in case, since we should never poll defi until onboarding is completed (this state does not exist in mobile though).